code-challenges.test.js

```diff
@@ -12,41 +12,158 @@
+//declare variable 
+//initialaze 
+//takes in an array
+//itterate through each value within the array
+//identify a particular letter within the values -> declare the letter as a variable
+//return the identified value if it includes the specified letter
 
-const secretCodeWord1 = "Lackadaisical"
-// Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
-// Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
 
-// b) Create the function that makes the test pass.
-
-// --------------------2) Create a function that takes in an array of words and a single letter and returns an array of all the words containing that particular letter.
 
 // a) Create a test with expects statement using the variable provided.
 
-const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+describe('filterWordsByLetter', () => {
+    it('returns an array of words containing the specified letter', () => {
+      const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"];
+      const filterLetterA = "a";
+      const filterLetterE = "e";
+  
+      expect(filterWordsByLetter(fruitArray, filterLetterA)).toEqual(["Mango", "Apricot", "Peach"]);
+      expect(filterWordsByLetter(fruitArray, filterLetterE)).toEqual(["Cherry", "Blueberry", "Peach"]);
+    });
+  });
 
-const filterLetterA = "a"
-// Expected output: ["Mango", "Apricot", "Peach"]
-const filterLetterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
+  //  FAIL  ./code-challenges.test.js
+//   ReferenceError: filterWordsByLetter is not defined
 
 // b) Create the function that makes the test pass.
 
+// ********************************************BELOW DID NOT WORK**********************************************************
+// const filterWordsByLetter = () => {
+//     const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"];
+//     //initializing the empty array for each expect
+//     const filterLetterAArray = [];
+//     const filterLetterEArray = [];
+//     // interating through, starting at 0 index through the length of the fruitArray variable array from Left to Right. 
+//     for (let i = 0; i < fruitArray.length; i++) {
+//         if (fruitArray[i].includes("a") || fruitArray[i].includes("A")) {
+//           filterLetterAArray.push(fruitArray[i]);
+//         } else if (fruitArray[i].includes("e") || fruitArray[i].includes("E")) {
+//           filterLetterEArray.push(fruitArray[i]);
+//         }
+//       }
+    
+//       return [filterLetterAArray] &&  [filterLetterEArray];
+//     }
+// ********************************************ABOVE DID NOT WORK**********************************************************
+
+const filterWordsByLetter = (fruitArray, filterLetter) => {
+    //initialize an empty array to input passing values
```

Comment:
Great pseudo code again. I would challenge you to a refactor - since we are working on an array and want an array as our output, what higher order function can we use for iteration to clean this up?